### PR TITLE
[Merged by Bors] - chore(Data/Int/Cast): fix confusion between `OfNat` and `Nat.cast` lemmas

### PR DESCRIPTION
**Pull request by eric-wieser**
This renames

* `Int.cast_ofNat` to `Int.cast_natCast`
* `Int.int_cast_ofNat` to `Int.cast_ofNat`

I think the history here is that this lemma was previously about `Int.ofNat`, before we globally fixed the simp-normal form to be `Nat.cast`.

Since the `Int.cast_ofNat` name is repurposed, it can't be deprecated. `Int.int_cast_ofNat` is such a wonky name that it was probably never used.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Int/Cast/Basic.lean [hunk 1]
```diff
@@ -61,30 +61,30 @@ theorem cast_zero : ((0 : ℤ) : R) = 0 :=
...
-@[simp high, nolint simpNF, norm_cast] -- this lemma competes with `Int.ofNat_eq_cast` to come later
-theorem cast_ofNat (n : ℕ) : ((n : ℤ) : R) = n :=
+@[simp, norm_cast]
+theorem cast_natCast (n : ℕ) : ((n : ℤ) : R) = n :=
   AddGroupWithOne.intCast_ofNat _
-#align int.cast_coe_nat Int.cast_ofNatₓ
+#align int.cast_coe_nat Int.cast_natCastₓ
 -- expected `n` to be implicit, and `HasLiftT`
-#align int.cast_of_nat Int.cast_ofNatₓ
+#align int.cast_of_nat Int.cast_natCastₓ
```
Comment (YaelDillies) [inline Mathlib/Data/Int/Cast/Basic.lean:69]: Should that be
```suggestion
#align int.cast_of_nat Int.cast_ofNat
```
(maybe with an `ₓ`?

File: Mathlib/Data/Num/Lemmas.lean [hunk 2]
```diff
@@ -1536,7 +1536,7 @@ theorem of_int_cast [AddGroupWithOne α] (n : ℤ) : ((n : ZNum) : α) = n := by
 
 @[simp, norm_cast]
 theorem of_nat_cast [AddGroupWithOne α] (n : ℕ) : ((n : ZNum) : α) = n := by
-  rw [← Int.cast_ofNat, of_int_cast, Int.cast_ofNat]
+  rw [← Int.cast_natCast, of_int_castcast_natCastCast]
```
Comment (YaelDillies) [inline Mathlib/Data/Num/Lemmas.lean:1539]: You suffered from ilean caching here

**Review by YaelDillies — COMMENTED** [review]
Could you also rename `AddGroupWithOne.intCast_ofNat`? It refers to `Nat.cast`, I just checked.

**Comment by eric-wieser** [comment]
> Could you also rename `AddGroupWithOne.intCast_ofNat`? It refers to `Nat.cast`, I just checked.

Let's do that in a separate PR, to avoid picking up more conflicts.

File: Mathlib/Data/Int/Cast/Basic.lean (hunk 1 above)
Comment (eric-wieser) [reply Mathlib/Data/Int/Cast/Basic.lean:70]: No, this is correct. The `cast_of_nat` lemma in lean 3 is morally the same as `cast_natCast`; both are about coercions not literals.

The lean3 version of `cast_ofNat` is `cast_bit0` / `cast_bit1`.

**Comment by j-loreaux** [comment]
Looks good, thanks!

bors merge

**Pull request closed without merging**